### DISCUSSION TITLE
Support "link" as a valid dependency version in package.json

### DISCUSSION THREAD

**Discussion by ThisIsMissEm** [npm/feedback#667] [CLI]
Currently when you're working on a project wherein you deliberately need to force using `npm link` for a dependency, e.g., you want the project to be developed against a local build of a package, then currently you need to have something like this in your dependencies:

```
  "dependencies": {
    "@inrupt/solid-client": "*"
  }
```

Such that tooling doesn't complain about `@inrupt/solid-client` not being installed, however, using a `*` (star / asterisks) version for a package isn't really recommended (I've seen tweets from @isaacs or @seldo about this).

I'd like to propose being able to instead write:

```
  "dependencies": {
    "@inrupt/solid-client": "link"
  }
```

Which when you run `npm install`, it would automatically execute `npm link <pkg>` for those, and if that fails, then print an error letting you know how to use `npm link` from the local copy of the dependency.

This would also be more ergonomic in places where you've multiple packages that you need to resolve from links; if you're using `npm ci` then I think the expected behaviour would be that this would fail, where as the former with `*` would succeed.

My primary use case for this is from example apps and e2e test apps where I want to build an application that's run against my package's latest code, without using relative imports.

**Comment by ljharb** [comment]
I believe this already works with a relative file path - `file:./path/to/package`.

**Reply by ThisIsMissEm** [reply]
Whilst `file:` could work, I don't think that's quite the same, similar but not quite the same.

**Reply by ljharb** [reply]
It creates a symlink in npm 5+ (before that, it copied)

**Reply by ThisIsMissEm** [reply]
Oh, maybe that works better! Does it still use the package.json and ensure that you're only requiring from the exports map?

**Reply by ljharb** [reply]
Yes, that's something node handles, which doesn't care if it's a symlink or not.
